internal/lua-sandbox/api/api.go

```diff
@@ -0,0 +1,575 @@
+	lState.SetGlobal("goGetGitObjectSize", lState.NewFunction(goGetGitObjectSize))
+	lState.SetGlobal("goGetGitObjectHash", lState.NewFunction(goGetGitObjectHash))
+	lState.SetGlobal("goGetGitObjectPath", lState.NewFunction(goGetGitObjectPath))
+
+	if err := lState.DoString(luaHelpers); err != nil {
+		return nil, err
+	}
+
+	lState.SetGlobal("hookParameters", lua.LString(""))
+	lState.SetGlobal("hookExitCode", lua.LNumber(0))
+	lState.SetGlobal("allowedExecutables", lua.LString(strings.Join(allowedModules, ",")))
+
+	return lState, nil
+}
+
+// Return all filenames in the specified directory, take a second argument to
+// specify if the scan should be recursive
+func goScanDir(l *lua.LState) int {
+	dirPath := l.ToString(1)
+	recursive := l.ToBool(2)
+	if !sandbox.IsPathAllowed(dirPath) {
+		l.Push(lua.LString("Error: Access to this directory is not allowed"))
+		return 1
+	}
+
+	var files []string
+	// Recursively scan the directory
+	if recursive {
+		err := filepath.Walk(dirPath, func(path string, _ os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			files = append(files, path)
+			return nil
+		})
+		if err != nil {
+			l.Push(lua.LString(fmt.Sprintf("Error scanning directory: %s", err.Error())))
+			return 1
+		}
+	} else {
+		// Return all files in the current directory without entering subdirectories
+		fileInfo, err := os.ReadDir(dirPath)
+		if err != nil {
+			l.Push(lua.LString(fmt.Sprintf("Error scanning directory: %s", err.Error())))
+			return 1
+		}
+		for _, file := range fileInfo {
+			files = append(files, file.Name())
+		}
+	}
+
+	// Return all scanned files as a string to the Lua sandbox
+	l.Push(lua.LString(strings.Join(files, "\n")))
+	return 1
+}
+
+// goReadFile reads the content of a file and returns it as a string
+func goReadFile(l *lua.LState) int {
+	filePath := l.ToString(1)
+	if !sandbox.IsPathAllowed(filePath) {
+		l.Push(lua.LString("Error: Access to this file is not allowed"))
+		return 1
+	}
+
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		l.Push(lua.LString(fmt.Sprintf("Error reading file: %s", err.Error())))
+		return 1
+	}
+
+	l.Push(lua.LString(string(data)))
+	return 1
+}
+
+// goGetDiff retrieves the git diff output
+func goGetDiff(l *lua.LState) int {
+	cmd := exec.Command("git", "diff", "HEAD", "--no-ext-diff", "--unified=0", "-a", "--no-prefix")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		l.Push(lua.LString(err.Error()))
+		return 1
+	}
+
+	l.Push(lua.LString(string(output)))
+	return 1
+}
+
+// goGetWorkTree retrieves the list of files in the git work tree
+func goGetWorkTree(l *lua.LState) int {
+	cmd := exec.Command("git", "ls-files")
+	output, err := cmd.CombinedOutput()
+
+	if err != nil {
+		l.Push(lua.LString(err.Error()))
+		return 1
+	}
+
+	l.Push(lua.LString(string(output)))
+	return 1
+}
+
+// goRegexMatch processes input text and searches for patterns
+func goRegexMatch(l *lua.LState) int {
```

Comment:
we want to reason about all of these APIs carefully; they're a big part of what we allow and don't allow in the sandbox. I suggest having only the most basic ones in this PR like a split string, regex match, and move the others to a follow on PR.